.changeset/grumpy-hats-unite.md

```diff
@@ -0,0 +1,5 @@
+---
+'neverthrow': minor
```

Comment:
This might be considered a breaking change, since users that use the function like described in the PR will get red squiggles. However, I would argue that this is a "type bug", and shouldn't be considered a valid use case